core/src/components/cat-time/cat-time.tsx

```diff
@@ -234,22 +243,27 @@ export class CatTime {
   @Method()
   async select(date: Date | null): Promise<void> {
     const oldValue = this.value;
+    let newValue = this.value;
     if (!date) {
       this.selectionTime = null;
       this.value = undefined;
     } else {
       const time = clampTime(this.min ?? null, date, this.max ?? null);
       this.isAm = this.format(time).toLowerCase().includes('am');
       this.selectionTime = time;
-      this.value = formatIso(time);
+      newValue = formatIso(time);
     }
     // we need to set the input explicitly to sync the input even without a
     // rerender (if the value is not changed)
     if (this.input) {
       this.input.value = this.format(this.selectionTime, false);
     }
-    if (oldValue !== this.value) {
+    if (oldValue !== newValue) {
+      this.programmaticallyChanged = true;
```

Comment:
what exactly is this flag doing? The name does not really gives a hint.

Reply:
It is a flag to know if the value was changed inside the component or outside the component, it is true that the name could be improved, maybe `valueChangedInternally` or `valueChangedBySync`  or `valueChangedBySelection`

Reply:
changed